### PR TITLE
Fix Peering Issues in Discovery

### LINKED ISSUE #6324: Sudden drop of connected peers
# 🐞 Bug Report

### Description

Peer count suddenly drops to around 5-7 peers and never raises to 30 again.

image

It is running on a home line with a 24h disconnect where the ip is likely to change after reconnecting. Beacon-node is set to announce the ddns name with `--p2p-host-dns`

- 24h disconnect was around 3:20h that night, but peers started dropping at around 2:00h already
- On previous days the disconnect had only minimal effect and peer count raised to 30 immediately again.
- There are still inbound and outbound connections happening occasionally.

### Has this worked before in a previous version?

Never noticed similar problems.

## 🔬 Minimal Reproduction

Not tried to reproduce it yet.

## 🌍  Your Environment

**Operating System:**

Ubuntu 20.04 on arm64
Docker

**What version of Prysm are you running? (Which release)**

802bdf5 compiled from source

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] It updates our geth fork to not spam lookup requests to our other peers in our dht. This
prevents our node from potentially overloading the local machine through excessive outbound requests. 
- [x] Stops initiating new lookup requests once we are at our peer limit. Instead it opts to pause the main routine at intervals

**Which issues(s) does this PR fix?**

Potentially resolves the issues raised in #6324 

**Other notes for review**

### CODE REVIEW

File: beacon-chain/p2p/service.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ import (
 var _ = shared.Service(&Service{})
 
 // Check local table every 15 seconds for newly added peers.
-var pollingPeriod = 15 * time.Second
+var pollingPeriod = 6 * time.Second
```
Comment (terencechain) [inline beacon-chain/p2p/service.go:49]: is 6 chosen because it's half of slot?

Comment (nisdas) [reply beacon-chain/p2p/service.go:49]: Not really, its pretty arbitrary.

File: beacon-chain/p2p/discovery.go [hunk 2]
```diff
@@ -182,6 +173,15 @@ func (s *Service) filterPeer(node *enode.Node) bool {
 	return true
 }
 
+// check if peer is at max limit.
```
Comment (prestonvanloon) [inline beacon-chain/p2p/discovery.go:176]: Can you please add a bit more to this? What limit? What peer?

Comment (nisdas) [reply beacon-chain/p2p/discovery.go]: added comment

**Comment by rauljordan** [comment]
small conflict @nisdas

**Pull request merged**